server/src/main/java/org/candlepin/model/PoolCurator.java

```diff
@@ -830,4 +830,12 @@ public void calculateExportedForOwnersPools(Owner owner) {
         q.setParameter("owner", owner);
         q.executeUpdate();
     }
+
+    public void markCertificatesDirtyForProductId(String productId) {
+        String statement = "update Entitlement e set e.dirty=true where e.pool.id in " +
+            "(select p.id from Pool p where p.productId = :productId)";
```

Comment:
This won't catch every pool the ```listAvailableEntitlementPools``` will fetch; it needs to be also checking the provided product ID as well. See: [PoolCurator.java, Line 290](https://github.com/candlepin/candlepin/blob/c1ca0255d374feff750cc14d3542f7e377021c38/server/src/main/java/org/candlepin/model/PoolCurator.java#L290).